core/src/worker/workflow/machines/patch_state_machine.rs

```diff
@@ -656,4 +750,75 @@ mod tests {
 
         wfm.shutdown().await.unwrap();
     }
+
+    const SIZE_OVERFLOW_PATCH_AMOUNT: usize = 180;
+    #[rstest]
+    #[case::happy_path(50)]
+    // We start exceeding the 2k size limit at 180 patches with this format
```

Comment:
I wonder if it's worth an integration test to ensure server behavior (e.g. confirm they aren't using, say 2000 as a limit). We can do this lang side if needed.

Reply:
I read through the code and the 2k value limit is quite clear - in any case this is sort of a "best effort" since in theory users could change the value.

Reply:
:+1: On the .NET or Python side I may have an integration test that pushes SA past the limit.